content/2015.08.notifications-preamble.md

```diff
@@ -0,0 +1,166 @@
+
+No problem. We can do that.
+
+# But wait, there's more!
+
+Sure, Kinto is a Mozilla project - but it's meant for general use, which means that anything we design needs to be reasonably generic. With this in mind we committed to designing something that can be used in a variety of contexts; after all, we're an organisation built for and on the web, and as such, we're dedicated to using web-centric solutions.
+
+Thinking then about what sort of parameters a reasonably generic solution might entail lead us to a few basic guidelines for designing the notifications mechanism:
+
+* We should strive to be as *agnostic* as possible, meaning that locking into a particular service, back-end, or even programming language, should be avoided where possible.
+* Since we're web-centric, we should capitalise on HTTP wherever possible as well.
+* The notifications mechanism must be able to scale with a minimum of hardship; this applies to any eventual queues, workers, and so forth.
+* Finally, it must be robust - we can't be losing messages now, can we?
+
+# How?
+
+There are about as many ways to go about solving the problem as there are grains of sand on a beach (warning: previous statement may not be scientifically valid). The best way to determine the winning approach is simple: through single combat, [Thunderdome-style](https://duckduckgo.com/?q=beyond+thunderdome&ia=about). Unfortunately our manager put the kibosh on that plan for liability reasons, so instead we did the next best thing: we drew on a whiteboard.
+
+In the interest of absolutely transparency, here are some of the models that we worked through and our general observations on each.
+
+## The Simple
+
+One approach is to be as [hands-off](<https://i.imgur.com/xZQitlj.gif) as possible. The premise here is straightforward enough: Kinto drops notifications into a queue (like Redis or RabbitMQ). That's basically it.
+
+{% graphviz
+    dot {
+        digraph simple {
+            rankdir=LR;
+            Kinto [shape=box];
+            Queue [shape=box];
+            Consumer [shape=box];
+            Kinto -> Queue;
+            Queue -> Consumer [style=dotted];
+            Consumer -> Queue [style=dotted];
+        }
+    }
+%}
+
+Pros:
+
+* This model implies a highly limited scope insofar as Kinto is directly concerned; we would basically treat Kinto like a lightweight transport mechanism, and the queue as a fire-and-forget target.
+* The amount of additional code would be relatively small - in essence, only just enough to speak to a queue service and nothing more.
+* We could ship with support for a handful of external queuing services.
+
+Cons:
+
+* We'd have to code and maintain support for a handful of external queuing services.
```

Comment:
How about "Each queueing service would require a dedicated module to be coded and maintained; this would likely
be an exercise left up to the community as part of the larger ecosystem." :question: